scene/animation/skeleton_ik.cpp

```diff
@@ -331,6 +331,12 @@ void FabrikInverseKinematic::solve(Task *p_task, real_t blending_delta, bool ove
 				new_bone_pose.basis = new_bone_pose.basis * p_task->chain.tips[0].end_effector->goal_transform.basis;
 		}
 
+		// IK should not affect scale, so undo any scaling
+		new_bone_pose.basis[0].normalize();
+		new_bone_pose.basis[1].normalize();
+		new_bone_pose.basis[2].normalize();
```

Comment:
Is there a better way to remove only the scale from a Basis? Would the `orthonormalized` function work?

Reply:
You might have to handle the non-uniform scaling removal. I don't know if that's possible..

Reply:
I'm *hoping* it will not be an issue, as after normalizing I apply whatever scale is set in the global position, which should reapply any non-uniform scales. It could be an issue though, I had not thought to test this. For most cases though, it should (fingers crossed) be okay.

Reply:
You can use this: https://github.com/godotengine/godot/blob/master/core/math/basis.cpp#L74